azure-pipelines.yml

```diff
@@ -3,36 +3,79 @@
 # Add steps that build, run tests, deploy, and more:
 # https://aka.ms/yaml
 
-jobs:
-- job: macOS
-  pool:
-    vmImage: 'macOS-10.14'
-  steps:
-  - script: brew install sfml
-  - script: mkdir build
-  - script: cd build && cmake -DBUILD_TESTS=ON ..
-  - script: cd build && make -j6
-  - script: cd build && ./tests/ObEngineTests
- 
-- job: Windows
-  variables:
-    MSVC_DEFAULT_OPTIONS: ON
-    SFML_DIR: D:\a\1\s\Libraries\SFML-2.5.1
-  pool:
-    vmImage: 'windows-2019'
-  steps:
-  - task: DownloadFile@1
-    inputs:
-      FileUrl: 'https://www.sfml-dev.org/files/SFML-2.5.1-windows-vc15-64-bit.zip'
-      DestinationFolder: '$(System.DefaultWorkingDirectory)'
-      requireAuthentication: false
-  - task: ExtractFiles@1
-    inputs:
-      archiveFilePatterns: 'SFML-2.5.1-windows-vc15-64-bit.zip'
-      destinationFolder: 'Libraries'
-      cleanDestinationFolder: true
-  - script: cmake -G "Visual Studio 16 2019" -A x64 -DSFML_DIR=%SFML_DIR% -DBUILD_TESTS=ON .
-  - script: dir
-  - task: VSBuild@1
-    inputs:
-      solution: '**\*.sln'
+strategy:
+  matrix:
+    linux:
+      imageName: 'ubuntu-18.04'
+    mac:
+      imageName: 'macos-10.14'
+    windows:
+      imageName: 'windows-2019'
+
+variables:
+  MSVC_DEFAULT_OPTIONS: ON
+  SFML_DIR: D:\a\1\s\Libraries\SFML-2.5.1
+
+pool:
+  vmImage: $(imageName)
+
+steps:
+
+- task: DownloadFile@1
+  inputs:
+    FileUrl: 'https://www.sfml-dev.org/files/SFML-2.5.1-windows-vc15-64-bit.zip'
+    DestinationFolder: '$(System.DefaultWorkingDirectory)'
+    requireAuthentication: false
+  condition: eq( variables['Agent.OS'], 'Windows_NT' )
+  displayName: "[Windows] Download SFML file"
+- task: ExtractFiles@1
+  inputs:
+    archiveFilePatterns: 'SFML-2.5.1-windows-vc15-64-bit.zip'
+    destinationFolder: 'Libraries'
+    cleanDestinationFolder: true
+  condition: eq( variables['Agent.OS'], 'Windows_NT' )
+  displayName: "[Windows] Extract SFML file"
+
+- script: cmake -G "Visual Studio 16 2019" -A x64 -DSFML_DIR=%SFML_DIR% -DBUILD_TESTS=ON .
+  condition: eq( variables['Agent.OS'], 'Windows_NT' )
+  displayName: "[Windows] Cmake"
+- task: VSBuild@1
+  inputs:
+    solution: '**\*.sln'
+  condition: eq( variables['Agent.OS'], 'Windows_NT' )
+  displayName: "[Windows] VS"
+- script: |
+    brew install sfml
```

Comment:
Could you move the task up, where dependencies are being installed for other images ?